### PR TITLE
Add warning for manually determining core kubevirt infra component replica count

### PR DESCRIPTION
**Pull request by iholder-redhat**
**What this PR does / why we need it**:
[This PR](https://github.com/kubevirt/kubevirt/pull/6663) introduces an option in Kubevirt CR's spec to manually determine the number of replicas for Kubevirt's core components like `virt-api`, `virt-controller`, etc.

As laid out in [this discussion](https://github.com/kubevirt/kubevirt/pull/7397#discussion_r828858282), this is problematic as it prevents us from auto-scaling the component in a smart way. As we're heading for auto-piloting and auto-scaling our component (e.g. [this PR](https://github.com/kubevirt/kubevirt/pull/7401)), this field is not recommended for use, expect for maybe advanced users that know exactly what they're doing.

**Release note**:

```release-note
Add warning for manually determining core-component replica count in Kubevirt CR
```

### CODE REVIEW

File: pkg/virt-operator/resource/apply/apps.go [hunk 1]
```diff
@@ -54,6 +54,8 @@ func (r *Reconciler) syncDeployment(origDeployment *appsv1.Deployment) (*appsv1.
 		replicas := int32(*kv.Spec.Infra.Replicas)
 		if deployment.Spec.Replicas == nil || *deployment.Spec.Replicas != replicas {
 			deployment.Spec.Replicas = &replicas
+			log.Log.Warningf("applying custom number of infra replica. this is an advanced feature that prevents " +
```
Comment (xpivarc) [inline pkg/virt-operator/resource/apply/apps.go:57]: I am not sure how many users will be looking into logs. Maybe you want to use event?

**Review by enp0s3 — APPROVED** [review]
/lgtm

**Pull request merged**